### PR TITLE
feat(sdk-core): ecdsa signing use derivation path

### PR DESCRIPTION
**Pull request by olibeyene**
Ticket: BG-66641

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 1]
```diff
@@ -583,11 +583,14 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
...
-    const signingKey = MPC.keyCombine(userSigningMaterial.pShare, [
-      userSigningMaterial.bitgoNShare,
-      userSigningMaterial.backupNShare,
-    ]);
+    const keyDerive = MPC.keyDerive(
+      userSigningMaterial.pShare,
+      [userSigningMaterial.bitgoNShare, userSigningMaterial.backupNShare],
+      derivationPath
+    );
+    const signingKey = MPC.keyCombine(userSigningMaterial.pShare, [keyDerive.nShares[3], keyDerive.nShares[2]]);
```
Comment (johnoliverdriscoll) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:593]: Not necessary for the user to call `keyCombine` here. In fact, the signingKey will not produce valid signing shares for the expected key. The user already has their signing material as `keyDerive.xShare`.

**Review by johnoliverdriscoll — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 2]
```diff
@@ -612,18 +612,44 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
...
+      );
+      const unsignedTx =
+        txRequestResolved.apiVersion === 'full'
+          ? txRequestResolved.transactions![0].unsignedTx
+          : txRequestResolved.unsignedTxs[0];
+      signablePayload = Buffer.from(unsignedTx.signableHex, 'hex');
+      derivationPath = txRequestResolved.transactions![0].unsignedTx.derivationPath;
+    } else if (requestType === RequestType.message) {
+      signablePayload = (params as TSSParamsForMessage).bufferToSign;
+      // TODO Message singing with derivation path
```
Comment (olibeyene) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:631]: @jeanlarrieux  can you take look at what it takes to add derivation path to message signing?

**Review by wesgraham — APPROVED** [review]

File: modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts [hunk 3]
```diff
@@ -730,18 +730,19 @@ describe('TSS Ecdsa Utils:', async function () {
     });
 
     async function setupSignTxRequestNocks(isTxRequest = true) {
-      let response = { txRequests: [{ ...txRequest, transactions: [{ ...txRequest, unsignedTx: { signableHex: txRequest.unsignedTxs[0].signableHex, serializedTxHex: txRequest.unsignedTxs[0].serializedTxHex } }] }] };
+      const derivationPath = '';
```
Comment (jicli) [inline modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts:733]: ideally would love to see an example with actual derivation path test case

**Review by jicli — APPROVED** [review]

File: modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts (hunk 3 above)
Comment (sachushaji) [reply modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts:733]: +1

**Review by johnoliverdriscoll — APPROVED** [review]
lgtm, given that this does not actually transmit a derivation path, and simply provides a no-op parameter

File: modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts (hunk 3 above)
Comment (olibeyene) [reply modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts:733]: I will add additional test in follow up pr BG-67621

**Review by byhow — APPROVED** [review]

**Pull request merged**